mythril/laser/ethereum/instructions.py

```diff
@@ -1184,18 +1186,30 @@ def extcodecopy_(self, global_state: GlobalState) -> List[GlobalState]:
             global_state=global_state,
         )
 
-    @StateTransition
+    @StateTransition()
     def extcodehash_(self, global_state: GlobalState) -> List[GlobalState]:
         """
 
         :param global_state:
         :return: List of global states possible, list of size 1 in this case
         """
-        # TODO: To be implemented
-        address = global_state.mstate.stack.pop()
-        global_state.mstate.stack.append(
-            global_state.new_bitvec("extcodehash_{}".format(str(address)), 256)
-        )
+        world_state = global_state.world_state
+        stack = global_state.mstate.stack
+        address = Extract(159, 0, stack.pop())
+
+        if address.symbolic:
+            log.debug("unsupported symbolic address for EXTCODEHASH")
+            stack.append(global_state.new_bitvec("extcodehash_" + str(address), 256))
+            return [global_state]
+
+        if address.value not in world_state.accounts:
+            code_hash = symbol_factory.BitVecVal(0, 256)
```

Comment:
This might not be needed. If I recall correctly, `world_state.accounts_exist_or_load` should add the account to the world_state if it is not already there.

Reply:
Hey Nathan, thanks for the feedback. For extcodehash, addresses that do not exist should return a 0, so I had to explicitly check it. I use the accounts_exist_or_load primarily to load the code of accounts that do exist.